IrohaCrypto/Classes/sr25519/SS58AddressFactory.m

```diff
@@ -49,12 +66,28 @@ - (nullable NSString*)addressFromPublicKey:(id<IRPublicKeyProtocol> _Nonnull)pub
     return [addressData toBase58];
 }
 
+- (UInt8) decodeTypeFromData:(NSData *)addressData {
+    UInt8 first = ((uint8_t*)addressData.bytes)[0];
+    switch (first) {
+    case 0 ... 63:
+        return first;
+    case 64 ... 127:;
+        UInt8 second = ((uint8_t*)addressData.bytes)[1];
+        UInt8 lower = first << 2 | second >> 6;
+        UInt8 upper = second & 0b00111111;
+        return lower | (upper << 8);
+    default:
+        break;
+    }
+    return 0;
+}
+
 - (nullable NSData*)accountIdFromAddress:(nonnull NSString*)address
-                                    type:(SNAddressType)type
+                                    type:(UInt8)type
                                    error:(NSError*_Nullable*_Nullable)error {
     NSData *ss58Data = [[NSData alloc] initWithBase58String:address];
 
-    if ([ss58Data length] != ADDRESS_LENGTH) {
+    if ([ss58Data length] != ADDRESS_LENGTH && [ss58Data length] != ADDRESS_LENGTH + 1) {
```

Comment:
Let't rename ADDRESS_LENGTH -> SIMPLE_ADDRESS_LENGTH and also introduce FULL_ADDRESS_LENGTH